apps/src/sites/studio/pages/home/_homepage.js

```diff
@@ -34,6 +36,7 @@ function showHomepage() {
   store.dispatch(setValidGrades(homepageData.valid_grades));
   store.dispatch(setStageExtrasScriptIds(homepageData.stageExtrasScriptIds));
   store.dispatch(setAuthProviders(homepageData.providers));
+  store.dispatch(setShowVersionMenu(experiments.isEnabled(experiments.VERSION_MENU)));
```

Comment:
Do we actually need to push this experiment state into redux?  Experiments are already a global concept, and this redux state is consumed in exactly one place.

Reply:
I started to just check the experiment within the AssignmentVersionSelector component, but then I dimly remembered this can lead to headache when trying to configure unit tests and storybook entries. Is this perhaps less of a problem than I'm remembering?

I'd also be happy to keep this out of redux and pass it down through the react component chain directly. although it touches a lot of files, it's straight forward to do / undo and I don't mind adding lines of code that I know won't be burdening us for long.

Reply:
I don't think it needs to be passed down through the react component chain at all - I was suggesting `AssignmentVersionSelector` check the experiment directly, since that's the only component that makes decisions based on this experiment.

```jsx
// AssignmentVersionSelector.jsx
import experiments from '@cdo/apps/util/experiments';
export default class AssignmentVersionSelector extends Component {
  showVersionMenu = experiments.isEnabled(experiments.VERSION_MENU);
  handleMouseDown = e => this.showVersionMenu && e.preventDefault();
  //...
}
```

I haven't had an issue with testing or storybooks with experiments, but YMMV. Here are the patterns I've used for working with experiments in unit tests:

https://github.com/code-dot-org/code-dot-org/blob/ec5778aa3c97502c132c1b3ee8447bd83f0ce752/apps/test/unit/lib/kits/maker/CircuitPlaygroundBoardTest.js#L66-L69

And in storybook:

https://github.com/code-dot-org/code-dot-org/blob/f6f57dfe1196d18c9c31fe5e8d9d5d4a654f3b81/apps/src/templates/projects/SectionProjectsList.story.jsx#L31-L34

The latter is [our own storybook extension](https://github.com/code-dot-org/code-dot-org/blob/f6f57dfe1196d18c9c31fe5e8d9d5d4a654f3b81/apps/.storybook/config.js#L117-L121).

Definitely not a blocking issue though - if you'd prefer to manage the experiment another way, I won't complain.